.github/workflows/build.yaml

```
@@ -0,0 +1,170 @@
+          ARTIFACT: entropy-${{ env.BUILD_VERSION }}-${{ matrix.os }}.jar
+          S3_KEY: ${{ env.GIT_SHA }}/${{ matrix.os }}/${{ env.ARTIFACT }}
+          S3_URI: s3://${{ env.S3_BUCKET }}/${{ env.S3_KEY }}
+        run: |
+          if aws s3api head-object --bucket "$S3_BUCKET" --key "$S3_KEY" 2>/dev/null; then
+            echo "BUILD_RESULT=skipped" >> $GITHUB_ENV
+            echo "✔️ Already exists—skipping build."
+          else
+            echo "BUILD_RESULT=built" >> $GITHUB_ENV
+            mvn clean package -P${{ matrix.os }} -Dbuild.revision="${BUILD_VERSION}"
+            aws s3 sync "target/$ARTIFACT" "$S3_URI"
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)